src/ashley/machina_extensions/forum/views.py

```diff
@@ -15,20 +17,132 @@
 
 Forum = get_model("forum", "Forum")
 
+ORDER_VAR = "o"
 PermissionRequiredMixin: BasePermissionRequiredMixin = get_class(
     "forum_permission.viewmixins", "PermissionRequiredMixin"
 )
 
 
+class ForumView(BaseForumView):  # pylint: disable=too-many-ancestors
+    """Displays a forum and its topics."""
+
+    # header columns
+    list_display = ["subject", "posts_count", "views_count", "last_post_on"]
+    # default ordering
+    ordering = ["-last_post_on"]
+
+    def get_default_index_order(self):
+        """Utility to access index query column and column of default order with sign"""
+        field = self.ordering[0].replace("-", "")
+        sign = "-" if "-" in self.ordering[0] else ""
+        return {
+            "index": f"{sign}{self.list_display.index(field)}",
+            "col": self.ordering[0],
+        }
+
+    def __init__(self, **kwargs):
+        super().__init__()
+        self.params = {}
+
+    def get(self, request, **kwargs):
+        """Handles GET requests."""
+        # collect the current params for the order by in request
+        self.params = dict(request.GET.items())
+        response = super().get(request, **kwargs)
+        return response
+
+    def get_default_ordering(self):
+        """Gets order requested in url"""
+        ordering = self.request.GET.get(
+            ORDER_VAR, self.get_default_index_order().get("index")
+        )
+        try:
+            _none, pfx, idx = ordering.rpartition("-")
+            # returns default column if out of index order
+            if (int)(idx) >= len(self.list_display):
+                idx = self.get_default_index_order().get("col")
+            return pfx, (int)(idx)
+        except ValueError:
+            # param exists but value is unexpected, use default order
+            _none, pfx, idx = (
+                self.get_default_index_order().get("index").rpartition("-")
+            )
+            return pfx, (int)(idx)
+
+    def get_queryset(self):
+        """Returns the list of items for this view ordered by asked param."""
+        query = super().get_queryset()
+        return query.order_by(self.get_ordering())
+
+    def get_context_data(self, **kwargs):
+        """Returns the context data to provide to the template."""
+        context = super().get_context_data(**kwargs)
+        # add sort header links containing asc, desc orders
+        context["header"] = list(self.result_headers())
+        # used for pagination to keep the order
+        if self.request.GET.get(ORDER_VAR):
+            context["order"] = f"{ORDER_VAR}={self.request.GET.get(ORDER_VAR)}"
```

Comment:
The value injected in the context without validation.
If an invalid value is provided, it will still be used to build pagination links, as you can see in the following example:

![Screenshot from 2021-04-21 15-40-59](https://user-images.githubusercontent.com/2926/115563815-331aab00-a2b8-11eb-85be-58bfca452385.png)

Reply:
True, I didn't think it could be a problem, but you right why allowing it! thanks